draft-specifications/Formats.md

```diff
@@ -0,0 +1,248 @@
+
+- _presentation-string_: `"` _presentation-character_ * `"`
+- _presentation-character_: _any printable ASCII character except `"` or `\`_ ;
+  note that spaces are printable
+- _wire-string_: _length_ `"` _bytes_
+- _length_: _integer-digits_ ; The number of bytes in _bytes_ as ASCII decimal
+  digits.
+- _bytes_: _byte_* ; The bytes of the string in UTF-8 encoding.
+
+We do not attempt to capture strings with embedded quotes or non-ASCII Unicode
+characters in the Presentation Format, but all Unicode strings excluding
+surrogates (U+D800-U+DFFF) are expressible in the Wire Format.
+
+## Selector
+
+A sequence of Unicode code points excluding surrogates (U+D800-U+DFFF).
+
+> Example: `'fleur-de-lis` corresponds to `12'fleur-de-lis`.
+
+- _presentation-selector_: `'` _name_
+- _name_: _alpha_ ( _alpha_ / _digit_ / `-` / `:` )*
+- _alpha_: ( `a` - `z` ) / ( `A` - `Z` )
+- _digit_: `0` - `9`
+- _wire-selector_: _length_ `'` _bytes_
+- _bytes_: _byte_* ; The bytes of the selector in UTF-8 encoding.
+
+We do not attempt to capture selectors with arbitrary Unicode characters in the
+Presentation Format, but all Unicode strings excluding surrogates
+(U+D800-U+DFFF) are expressible in the Wire Format.
+
+## ByteArray
+
+An array of 8-bit bytes.
+
+> Example: `:b0b5c0ffeefacade` corresponds to the bytes of `8`, `:`, b0, b5,
+> c0, ff, ee, fa, ca, de in ASCII and hexadecimal.
+
+- _presentation-byte-array_: `:` _hex_
+- _hex_: ( _hex-digit_ _hex-digit_ )*
+- _hex-digit_: _digit_ / ( `a` - `f` ) ; Corresponding to a _byte_ of _bytes_.
+- _wire-byte-array_: _length_ `:` _bytes_ ; The number of bytes in _bytes_
+  and the _bytes_.
+
+We do not attempt to capture byte arrays with non-space ASCII characters in the
+Presentation Format, but all byte arrays are expressible in the Wire Format.
+
+## Struct
+
+> The name "struct" is tentative.
+> https://github.com/ocapn/ocapn/pull/125
+
+A collection of (field name, value) pairs.
+
+In the Presentation Format and Wire Format, no field name may appear more than
+once.
+
+In the Wire Format, fields must appear in strictly-ascending bytewise order.
+
+> Examples:
+> - `{ a: 10, b: 2 }` corresponds to `{1"a10+1"b2+}`.
+> - `{ "a": 10, "b": 2 }` corresponds to `{1"a10+1"b2+}`.
+> - `{ 'b: 2, 'a: 10 }` corresponds to `{1'a10+1'b2+}`.
+
+- _presentation-struct_: `{` ( _presentation-field_ ( `,` _presentation-field_ )* )? `}`
+- _presentation-field_: _presentation-key_ `:` _presentation-value_
+- _presentation-key_: _presentation-field-name_ / _presentation-value_
+- _presentation-field-name_: _name_ ; Corresponding to a string.
+- _wire-struct_: `{` _wire-field_* `}`
+- _wire-field_: _wire-key_ _wire-value_
+- _wire-key_: _wire-value_ ; The Wire Format lacks an analog for the
+  Presentation Format's string key shorthand.
+
+> The [Model](Model.md) limits field names in structs to strings, but for
+> purposes of [CapTP](CapTP%20Specification.md) surrounding data, the notation
+> and representation allow any value.
```

Comment:
> the notation and representation allow any value.

Is "the notation" the presentation format?

Is "the representation" the wire format?

Reply:
I missed one in the translation. Thank you for catching it. Indeed, [Abstract] Notation became Presentation Format and [Concrete] Representation became Wire Format.